File: Euclid/Game/Room/Managers/RoomTaskManager.cs
using System.Collections.Generic;

namespace Euclid.Game
{
    public class RoomTaskManager : ILoadable
    {
        #region Fields

        private Room room;

        #endregion

        #region Properties

        public List<IRoomTask> Tasks { get; }

        #endregion

        #region Constructors

        public RoomTaskManager(Room room)
        {
           this.room = room;
           this.Tasks = new List<IRoomTask>();
        }

        #endregion

        #region Public methods

        /// <summary>
        /// Start all registered tasks
        /// </summary>
        public void Load()
        {
            StopTasks();
            Tasks.Clear();
            RegisterTasks();

            Tasks.ForEach(x => x.CreateTask());
        }

        /// <summary>
        /// Register all tasks
        /// </summary>
        private void RegisterTasks()
        {
            Tasks.Add(new EntityTask(room));
            //Tasks.Add(new StatusTask(room));
            //Tasks.Add(new ItemTickTask(room));
        }

        /// <summary>
        /// Stop all registered tasks
        /// </summary>
        public void StopTasks()
        {
            foreach (var task in Tasks)
                task.StopTask();
        }

        /// <summary>
        /// Convert real time to Habbo tick processing time
        /// </summary>
        public static int GetProcessTime(double time)
        {
            return (int)(time / 0.5);
        }

        #endregion
    }
}

File: euclid/src/main/java/euclid/game/room/RoomTaskManager.java
package euclid.game.room;

import euclid.game.ILoadable;
import euclid.game.room.tasks.EntityTask;
import euclid.game.room.tasks.IRoomTask;

import java.util.ArrayList;
import java.util.List;

public class RoomTaskManager implements ILoadable {

    private final Room room;
    private final List<IRoomTask> tasks;

    public RoomTaskManager(Room room) {
        this.room = room;
        this.tasks = new ArrayList<>();
    }

    public List<IRoomTask> getTasks() {
        return tasks;
    }

    @Override
    public void load() {
        stopTasks();
        tasks.clear();
        registerTasks();

        for (IRoomTask task : tasks)
            task.createTask();
    }

    private void registerTasks() {
        tasks.add(new EntityTask(room));
    }

    public void stopTasks() {
        for (IRoomTask task : tasks)
            task.stopTask();
    }

    public static int getProcessTime(double time) {
        return (int) (time / 0.5);
    }
}
